src/format/Format.cc

```diff
@@ -472,15 +472,9 @@ Format::Format::assemble(MemBuf &mb, const AccessLogEntry::Pointer &al, int logS
 
         case LFT_LOCAL_LISTENING_IP: {
             // avoid logging a dash if we have reliable info
-            const bool interceptedAtKnownPort = al->request ?
-                                                (al->request->flags.interceptTproxy ||
-                                                 al->request->flags.intercepted) && al->cache.port :
-                                                false;
-            if (interceptedAtKnownPort) {
-                const bool portAddressConfigured = !al->cache.port->s.isAnyAddr();
-                if (portAddressConfigured)
-                    out = al->cache.port->s.toStr(tmp, sizeof(tmp));
-            } else if (al->tcpClient)
+            if (const auto addr = FindListeningPortAddress(al->request))
+                out = addr->toStr(tmp, sizeof(tmp));
+            else if (al->tcpClient) // XXX: is this required anymore?
```

Comment:
We should try to figure out the answer to that question. AFAICT, the same or similar question can be phrased like this:

* Is it possible for `al->tcpClient` to be set when `al->request` or `al->request->masterXaction->squidPort` is not set?

I suspect the answer is "yes" because the request structure may even not exist for some logged events.

If my suspicion is correct, then we may even want to move this code and add AccessLogEntryPointer as the second FindListeningPortAddress() parameter so that future callers do not forget to supply it (callers that do not have it will supply nullptr). However, I do not insist on this change -- adding a comment to the ErrorPage caller is sufficient for now:

```
// TODO: When/if we get ALE here, use al->tcpClient as well
if (const auto addr = FindListeningPortAddress(request.getRaw()))
```

Reply:
Yes I suspected the same that al->request may not be set but al->tcpClient may be. Hence the XXX: comment.

Reply:
Added ability to pass al->cache.port->s and al->tcpClient->local as parameters. They will take precedence over analogues IPs obtainable via request->masterXaction. This should resolve above issues. Please check.